proof/records.go

```diff
@@ -362,6 +364,59 @@ func MetaRevealDataRecord(data *[]byte) tlv.Record {
 	)
 }
 
+func MetaRevealDecimalDisplayRecord(
+	decimalDisplay *fn.Option[uint32]) tlv.Record {
```

Comment:
Why a _pointer_ to the option vs just the option?

Reply:
Those `...Record()` functions are also used for decoding. So we need to be able to **write** to the parameter, that's why it's always a pointer.
We should upgrade all those records to the new TLV pattern (using `tlv.RecordT`), but that's a bigger refactor that should be its own PR.